### PR TITLE
Add button component and test

### PR DESCRIPTION
**Pull request by jiggsaw85**
**Jira task:** [https://reblaze.atlassian.net/jira/software/projects/V5GA/boards/39?selectedIssue=V5GA-19](https://reblaze.atlassian.net/jira/software/projects/V5GA/boards/39?selectedIssue=V5GA-19)

### CODE REVIEW

File: src/components/FormButton.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
+<template>
+  <button
+    :disabled="disabled"
+    class="button"
+    :class="[variant, display, icon ? 'icon' : '']"
+  >
+    <img v-if="icon && variant !== 'negative'" width="18" :src="icon" />
```
Comment (Aviv-Galmidi) [inline src/components/FormButton.vue:7]: Please separate each attributes by a new line

File: src/components/FormButton.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    <svg-icon v-if="variant === 'negative'" type="mdi" :path="mdiTrashCanOutline" size="18" />
```
Comment (Aviv-Galmidi) [inline src/components/FormButton.vue:8]: Please separate each attributes by a new line

File: src/components/FormButton.vue [hunk 3]
```diff
@@ -0,0 +1,180 @@
+<template>
+  <button
+    :disabled="disabled"
+    class="button"
+    :class="[variant, display, icon ? 'icon' : '']"
+  >
```
Comment (Aviv-Galmidi) [inline src/components/FormButton.vue:6]: Please keep the closing tag in the same line as the last attribute

File: src/components/FormButton.vue [hunk 4]
```diff
@@ -0,0 +1,180 @@
+<template>
+  <button
+    :disabled="disabled"
```
Comment (Aviv-Galmidi) [inline src/components/FormButton.vue:3]: Please keep the first attribute line in the same line as the HTML element

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

File: src/components/FormButton.vue (hunk 4 above)
Comment (jiggsaw85) [reply src/components/FormButton.vue]: @Aviv-Galmidi All fixed and storybook added.

**Comment by jiggsaw85** [comment]
@Aviv-Galmidi I have adapted button's variations.
If you want to use icon inside the button in order to test, you can put any image url inside "icon" property in storybook. I have tested with this randomly selected image: https://www.adobe.com/express/create/media_127a4cd0c28c2753638768caf8967503d38d01e4c.jpeg?width=400&format=jpeg&optimize=medium

**Comment by jiggsaw85** [comment]
Just a quick note... Following Figma (and that is how I work), image inside all button variations is 18px width but on negative button, it is 16px width.
Also, padding-left (distance from left button side to image/icon) is set to 15px but on negative button it is 16px.

These are details I am following and do not want to change it by my own.

File: src/views/help/HelpAndSupport.vue [hunk 5]
```diff
@@ -53,15 +53,23 @@
...
 
 <script lang="ts">
 import {defineComponent} from 'vue'
+import FormButton from '@/components/FormButton.vue'
 
 export default defineComponent({
   name: 'HelpAndSupport',
+  components: {
+    FormButton,
+  },
```
Comment (Aviv-Galmidi) [inline src/views/help/HelpAndSupport.vue:72]: Please remove the buttons from this page

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]
1. IMG icons

> If you want to use icon inside the button in order to test, you can put any image url inside "icon" property in storybook. I have tested with this randomly selected image: https://www.adobe.com/express/create/media_127a4cd0c28c2753638768caf8967503d38d01e4c.jpeg?width=400&format=jpeg&optimize=medium

The idea was to use SVG icons similar to the trashcan (but configurable) so let's say we want a `plus` SVG icon inside `New` button

This will also resolve your next question regarding sizes

> image inside all button variations is 18px width but on negative button, it is 16px width.

Similar to the trashcan setting a size to 16 similar to the trashcan would

2. Padding

> Also, padding-left (distance from left button side to image/icon) is set to 15px but on negative button it is 16px.

Let's set them to be both 16px to look the same

3. IMG icon alignment
Let's align the trashcan with the text inside the button

Screen Shot 2023-03-15 at 15 17 52

File: src/components/FormButton.vue [hunk 6]
```diff
@@ -2,10 +2,10 @@
   <button :disabled="disabled"
     class="button"
     :class="[variant, display, icon ? 'icon' : '']">
```
Comment (Aviv-Galmidi) [inline src/components/FormButton.vue:4]: `display` is not being set anywhere

```suggestion
    :class="[variant, icon ? 'icon' : '']">
```

**Review by Aviv-Galmidi — CHANGES_REQUESTED** [review]

**Comment by ljubebiza** [comment]
I have fixed redundancy

**Pull request merged**